### ISSUE TITLE
Gauge isn't correctly rendered when first value is 0

### ISSUE THREAD

**Issue by tjsousa** [bernii/gauge.js#17]
I found out that when the Gauge is initialized with a 0 value it is rendered like this:

![Not initialized](https://www.evernote.com/shard/s20/sh/12a05842-8cec-41f2-87fd-b91439b3ce17/5205d4d98ae8dfde51096a3602848f83/res/da317387-3c3c-4ad0-b33a-3a2398185d77/skitch.png?resizeSmall&width=832)

But if I initialize it with another value and the set it to 0, it shows like this:

![Initialized](https://www.evernote.com/shard/s20/sh/4c643a00-18c6-4908-966b-c65844ac2bad/6810921001363181a3c1b7465b09ab7f/res/5e34b9b9-066b-4ee3-8fc2-70dea90f8f71/skitch.png?resizeSmall&width=832)

Is there any way to initialize it to show like this without having to set it first to a non-zero value?

**Comment by bernii** [comment]
Thanks for spotting this! Fixed :-) (update you gauge.min.js)

**Comment by tjsousa** [comment]
Hey Bernard.

Firstly, thank you for your work with gauge.js, it's awesome. And double thanks for responding to this issue so quickly.

Unfortunately, although the pointer is now rendered correctly when initialized with 0, neither the background color nor the line width are initilialized with the values I provide:

![Something's missing](https://www.evernote.com/shard/s20/sh/165f886f-83e2-4e26-b75b-b231509bd2f0/f33d659ae8ce7098ac51d79ada354697/res/6851ad72-0587-4f50-a504-16bb44311528/skitch.png?resizeSmall&width=832)

I only get the correct 0 value render (as in the issue description above) if I firstly set it to a non-zero value (like 0.1)

**Comment by bernii** [comment]
yuck, that's right. Sorry for missing it! Fixed :-)

**Comment by tjsousa** [comment]
No problem. Actually, I think I've spotted yet a small problem but it's almost there ;)

![Something there](https://www.evernote.com/shard/s20/sh/58b89ec6-0389-4577-8fb0-4d67b656b1c3/fcf22a93ec8b65f5527e3545711f3870/res/4017c30e-193a-446b-9705-335045cdf1f6/skitch.png?resizeSmall&width=832)

**Comment by bernii** [comment]
Could you share your settings for the gauge so I can reproduce that?

**Comment by tjsousa** [comment]
Sure, these are the options I'm initializing the gauge with:

``` javascript
    this.options = {
        maxValue: 100,
        animationSpeed: 32,
        lines: 19,
        angle: 0.15,
        lineWidth: 0.34, // The line thickness
        pointer: {
          length: 1, // The radius of the inner circle
          strokeWidth: 0.027, // The rotation offset
          color: '#333' // Fill color
        },
        colorStart: '#0F6E38',   // Colors
        colorStop: '#0ACF38',    // just experiment with them
        strokeColor: '#B81A1A',   // to see which ones work best for you
        generateGradient: true
    };
```

And here the initialization code:

``` javascript
  this.gauge = new Gauge(canvas);
  this.gauge.setTextField(text);
  this.gauge.setOptions(this.options); // create sexy gauge!
  this.gauge.maxValue = this.options.maxValue; // set max gauge value
  this.gauge.animationSpeed = this.options.animationSpeed; // set animation speed (32 is default value)
  this.gauge.set(0);
```

**Comment by bernii** [comment]
What browser/OS combo are you using? Are you sure you've got the most recent version of gauge.min.js? I was trying to recreate you issue locally but without luck :( Give me some insight into your setup and I'll be able to fix that.

**Comment by tjsousa** [comment]
I'm using Chrome 24 on Mac OS X 10.8.2, and have updated to the latest version of gauge.min.js.

My setup consists of initializing the gauge with the values I provided, then periodically update its value ranging from 0 to 100.

I can try to extract the problem to a simple test scenario if that might help you out.

**Comment by bernii** [comment]
Thanks for your patience, fixed: (your scenario here: http://jsfiddle.net/berni/Sjrt9/ ). Please update your gauge.min.js :-)

**Comment by tjsousa** [comment]
Hi !

As I think it is related, I'm posting it here: just found out that there seems to be a similar problem with the Donut when initialized to zero.

Here's the scenario: http://jsfiddle.net/yDR6j/

When initialized with 0 the text isn't rendered, but if you set it to another value and then back to 0, it displays OK.

**Comment by marcogigliarano** [comment]
Is this bug fixed? I have solved with a double setting:  set(1) and just after set(0).

**Comment by kplindegaard** [comment]
@kibray: Just played with the jsfiddle mentioned above, and it's fixed for Gauge but not Donut.

**Comment by kplindegaard** [comment]
@tjsousa and @kibray: It's finally fixed for Donut in version 1.3.4. Gauge has been OK for a while.

**Comment by ilyaDegtyarenko** [comment]
**Yesterday I took a fresh script from the official site and I have this problem.**

![2019-01-06 19 03 48](https://user-images.githubusercontent.com/37239202/50739346-a8e1d200-11e7-11e9-975a-aa585382f181.png)
![2019-01-06 19 12 04](https://user-images.githubusercontent.com/37239202/50739348-a97a6880-11e7-11e9-9028-9c7af3d251ad.png)
![2019-01-06 19 05 13](https://user-images.githubusercontent.com/37239202/50739347-a97a6880-11e7-11e9-914f-bed49eb2aa03.png)

P.S. With a whole zero the same thing.

**Placed in this order and everything worked fine.**
![2019-01-06 19 36 17](https://user-images.githubusercontent.com/37239202/50739535-a634ac00-11ea-11e9-8adb-76832f9cae2b.png)
